### PR TITLE
Updated .listen to contain oldState and newState

### PR DESCRIPTION
**Pull request by aspiiire**
Updated docs to work with the current Stable version

### CODE REVIEW

File: website/docs/concepts/reading.mdx [hunk 1]
```diff
@@ -302,8 +306,8 @@ class HomeView extends ConsumerWidget {
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    ref.listen<int>(counterProvider, (int count) {
-      print('The counter changed ${count}');
+    ref.listen<int>(counterProvider, (int previousCount, int newCount) {
```
Comment (rrousselGit) [inline website/docs/concepts/reading.mdx:309]: Previous values are nullable

**Comment by rrousselGit** [comment]
Thanks! Just a small issue and it's good to go

**Comment by aspiiire** [comment]
> Thanks! Just a small issue and it's good to go

Thank you, could you check now that everything is fine? thanks for your time

File: website/docs/concepts/reading.mdx [hunk 2]
```diff
@@ -542,8 +546,8 @@ It is possible to use `select` with `ref.listen` too:
 ```dart
 ref.listen<String>(
   userProvider.select((user) => user.name),
-  (String name) {
-    print('The user name changed $name');
+  (String previousName, String newName) {
```
Comment (rrousselGit) [inline website/docs/concepts/reading.mdx:549]:
```suggestion
  (String? previousName, String newName) {
```

File: website/docs/concepts/reading.mdx [hunk 3]
```diff
@@ -279,14 +279,18 @@ if the listened provider changes, using `ref.listen` will instead call a custom
 That can be useful to perform actions when a certain change happens, such that
 to show a snackbar when an error happens.
 
+The `ref.listen` method needs 2 positional arguments, the first one is the Provider and the second one is the callback function that we want to execute when the state change.
+
```
Comment (rrousselGit) [inline website/docs/concepts/reading.mdx:283]:
```suggestion
```

**Comment by rrousselGit** [comment]
LGTM thanks!

**Pull request merged**